packages/file_selector/file_selector_web/lib/file_selector_web.dart

```diff
@@ -0,0 +1,205 @@
+import 'dart:async';
+import 'dart:html';
+
+import 'package:file_selector_platform_interface/file_selector_platform_interface.dart';
+import 'package:file_selector_platform_interface/src/web_helpers/web_helpers.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter_web_plugins/flutter_web_plugins.dart';
+import 'package:meta/meta.dart';
+
+final String _kFileSelectorInputsDomId = '__file_selector_web_file_input';
+
+/// The web implementation of [FileSelectorPlatform].
+///
+/// This class implements the `package:file_selector` functionality for the web.
+class FileSelectorPlugin extends FileSelectorPlatform {
+  Element _target;
+  final FileSelectorPluginTestOverrides _overrides;
+
+  bool get _hasTestOverrides => _overrides != null;
+
+  /// Open file dialog for loading files and return a XFile
+  @override
+  Future<XFile> loadFile({
+    List<XTypeGroup> acceptedTypeGroups,
+    String initialDirectory,
+    String confirmButtonText,
+  }) async {
+    return (await _loadFileHelper(false, acceptedTypeGroups)).first;
+  }
+
+  /// Open file dialog for loading files and return a XFile
+  @override
+  Future<List<XFile>> loadFiles({
+    List<XTypeGroup> acceptedTypeGroups,
+    String initialDirectory,
+    String confirmButtonText,
+  }) async {
+    return _loadFileHelper(true, acceptedTypeGroups);
+  }
+
+  @override
+  Future<String> getSavePath({
+    List<XTypeGroup> acceptedTypeGroups,
+    String initialDirectory,
+    String suggestedName,
+    String confirmButtonText,
+  }) async =>
+      null;
+
+  /// Load Helper
+  Future<List<XFile>> _loadFileHelper(
+      bool multiple, List<XTypeGroup> acceptedTypes) {
+    final acceptedTypeString = _getStringFromFilterGroup(acceptedTypes);
+
+    final FileUploadInputElement element =
+        createFileInputElement(acceptedTypeString, multiple);
+
+    _target.children.clear();
+    addElementToContainerAndClick(_target, element);
+
+    return getFilesWhenReady(element);
+  }
+
+  /// Default constructor, initializes _target to a DOM element that we can use
+  /// to host HTML elements.
+  /// overrides parameter allows for testing to override functions
+  FileSelectorPlugin({
+    @visibleForTesting FileSelectorPluginTestOverrides overrides,
+  }) : _overrides = overrides {
+    _target = ensureInitialized(_kFileSelectorInputsDomId);
+  }
+
+  /// Registers this class as the default instance of [FileSelectorPlatform].
+  static void registerWith(Registrar registrar) {
+    FileSelectorPlatform.instance = FileSelectorPlugin();
+  }
+
+  void _verifyXTypeGroup(XTypeGroup group) {
+    if (group.extensions == null &&
+        group.mimeTypes == null &&
+        group.webWildCards == null) {
```

Comment:
What if these are not null, but they're empty arrays?

Reply:
I'm picturing this

`group = XTypeGroup(extensions: []);`

😐 😐 

Should I add this to XTypeGroup constructor assertion?

Reply:
I think an empty XTypeGroup doesn't make too much sense (but checking only for null is not enough, you also need to see if it's empty, right?)

Reply:
Yes makes sense now 👍 
We need this assertion in both places and include null and empty checks